### ISSUE TITLE
create a resource buff that enables reasonable storage support

### ISSUE THREAD

**Issue by gidden** [cyclus/cyclus#1125]
Creating an actual StorageFacility has proven difficult in the wild because tracking storage times of resources in a performant way is difficult. This could likely be alleviated by developing a buffer subclass specifically for this purpose. Of course, it should keep track of entry times of resources, but also should

* only pop available resources in Pop methods
* allow querying of available resources

anything else?

**Comment by gidden** [comment]
@rwcarlsen will likely have some ideas since he has the most experience with our current buffers

**Comment by gonuke** [comment]
@nuclearkatie - to the extent that you can interpret this issue - do you think your additions to Storage have fixed this?

**Comment by nuclearkatie** [comment]
No, I'm pretty sure my work hasn't and won't address this. My changes haven't touched the way we keep track of processing time or anything
